packages/flutter_tools/lib/src/compile.dart

```diff
@@ -577,7 +599,16 @@ class DefaultResidentCompiler implements ResidentCompiler {
     if (!_controller.hasListener) {
       _controller.stream.listen(_handleCompilationRequest);
     }
-
+    // Write `generated_main.dart` under the tool's owned directory.
+    // This ensures that the `lib` directory isn't polluted with generated files.
+    final Directory buildDir = _fileSystem.file(mainUri.path)
```

Comment:
The mainUri could be anywhere, so this isn't stable. I think the correct generated directory should be available on the FlutterProject

Reply:
Done